### ISSUE TITLE
Not able to replicate the pretrained model when training from scratch

### ISSUE THREAD

**Issue by NiranthS** [princeton-vl/pytorch_stacked_hourglass#27]
Thanks a lot for this awesome work.
I am trying to train the shg module on a custom dataset. Before that, I tried training from scratch on the mpii dataset but it was not working as expected. The loss is reducing from 0.17 to 0.0007 in about 2 epochs and is constant from thereon. Even the PCKh score on the model I trained from scratch is near 0.00. Could you please help me with this?

I think the problem might be because of the difference in some library version, could you also share the exact version of libraries you used for training. Thanks in advance.
@crockwell @jiang-du

**Comment by crockwell** [comment]
Hi NiranthS, what version of the model are you using? And batch size? On MPII? I recall when training that initial drop, but the loss should still decrease more (although not dramatically) if training properly. You may want to look what is going on with heatmap or keypoint outputs. A common error is that predictions are very close to 0,0 if something weird is happening to your training data. With a loss that low, I don't think your keypoint accuracy should be exactly 0.00 either.

**Comment by NiranthS** [comment]
I am using the 2HG model with batch size 8 on MPII dataset.
Yes the predictions are very close to zero.
I have directly run the train.py script in the repo. The only change was the batch size.

**Comment by crockwell** [comment]
I see. Curious. How many epochs did you train for? Are you using "-c" to properly load your model during test time? I'd recommend visualizing results during training as well. Also, looking at heatmap activations is a good place to start. I think there is a bug on your end because I recently used this code which seemed to be working.

**Comment by crockwell** [comment]
possibly versions of your scipy or cv2 or pytorch etc. are different which could be doing something strange? But this sounds like maybe something more basic is happening. BTW with batch size of 8 you could also try reducing the learning rate and may train for >100 epochs

**Comment by NiranthS** [comment]
I trained for around 90 epochs and l remember loading properly in the test.py.
Sure, I will try reducing the learning rate and try again for 100+ epochs and let you know what happens. Thank You.

**Comment by NiranthS** [comment]
@crockwell  I tried as you said, trained for 200+epochs, still not able to get some proper PCKh score. It is near 0.
I even tried training with the same PyTorch, scipy and OpenCV versions as yours, but it was the same. Can you help me with this?

**Comment by Louis-w1214** [comment]
@NiranthS Have you solve your problem? I have met the same problem too.

**Comment by NiranthS** [comment]
@Louis-w1214  in my case, there was sone problem with the CUDA packages/GPU. When I tried the same code on another system, it worked.

**Comment by Louis-w1214** [comment]
@NiranthS Can u tell me the CUDA&cuDNN versions on your system where the codes worked?

**Comment by NiranthS** [comment]
Cuda compilation tools, release 10.1, V10.1.105
I think the problem was in my GPU/CPU/local installation of CUDA, because with same version of CUDA on cloud platform, it was working fine

**Comment by Louis-w1214** [comment]
@NiranthS It is strange, I got the same version of CUDA installed by anaconda. Can you give the loss when training?

**Comment by NiranthS** [comment]
Sorry man, I do not have those files now.

**Comment by dkurzend** [comment]
I seem to have the same problem: when training I start at a loss of 0.17 and it goes down to 0.0007 and stays there.

On the test set I get 
Val PCK @, 0.5 , total : 0.0 , count: 44239
Tra PCK @, 0.5 , total : 0.0 , count: 4443
Val PCK @, 0.5 , ankle : 0.0 , count: 4234
Tra PCK @, 0.5 , ankle : 0.0 , count: 392
Val PCK @, 0.5 , knee : 0.0 , count: 4963
...

I tried using following environments:

1)
torch=1.11.0
cuda=10.2

2) (as specified in the readme)
torch=1.5.0
cuda=10.1

In both cases I get the same results.

How did you guys solve this problem? @NiranthS @Louis-w1214
